### ISSUE TITLE
standalone=true throws Error: Build failed with 1 error

### ISSUE THREAD

**Issue by rtritto** [vikejs/vike-node#38]
### Description

In `vike.config.ts`, enable `standalone` option to `vikeNode` plugin (`vike-node/plugin`).

Notes:
- `root` in `vite.config.ts` is `src`
- no error with `standalone: false` or omit `standalone` option

### Repro (repo: [template-vike-solid-daisyui-hono](https://github.com/templates-ecosystem/template-vike-solid-daisyui-hono) ~ branch: [build-standalone-fails](https://github.com/templates-ecosystem/template-vike-solid-daisyui-hono/tree/build-standalone-fails))

- `git clone https://github.com/templates-ecosystem/template-vike-solid-daisyui-hono.git`
- `git checkout react-not-defined`
- `cd template-vike-solid-daisyui-hono`
- `yarn`
- `yarn build` or `yarn vite build`

```console
vite v5.4.11 building for production...

(!) outDir C:\template-vike-solid-daisyui-hono\dist\client is not inside project root and will not be emptied.
Use --emptyOutDir to override.

✓ 149 modules transformed.
../dist/client/_temp_manifest.json                               1.73 kB │ gzip:  0.47 kB
../dist/client/assets/entries/pages_index.GJsVgUTz.js            1.14 kB │ gzip:  0.46 kB
../dist/client/assets/entries/pages_error.CqcNzHvx.js            1.41 kB │ gzip:  0.62 kB
../dist/client/assets/chunks/chunk-B15uGeP7.js                   4.55 kB │ gzip:  2.06 kB
../dist/client/assets/chunks/chunk-C0-CiukQ.js                  23.11 kB │ gzip:  8.96 kB
../dist/client/assets/entries/entry-client-routing.WR2FyNOe.js  69.12 kB │ gzip: 22.29 kB
vite v5.4.11 building SSR bundle for production...

(!) outDir C:\template-vike-solid-daisyui-hono\dist\server is not inside project root and will not be emptied.
Use --emptyOutDir to override.

✓ 7 modules transformed.
../dist/server/package.json             0.02 kB
../dist/server/_temp_manifest.json      0.81 kB
../dist/server/entries/pages_index.mjs  1.27 kB
../dist/server/entries/pages_error.mjs  1.95 kB
../dist/server/index.mjs                3.78 kB
✓ built in 47ms
X [ERROR] Could not resolve "C:/template-vike-solid-daisyui-hono/src/C:/template-vike-solid-daisyui-hono/dist/server/index.mjs"

Error: Build failed with 1 error:
error: Could not resolve "C:/template-vike-solid-daisyui-hono/src/C:/template-vike-solid-daisyui-hono/dist/server/index.mjs"
    at failureErrorWithLog (C:\template-vike-solid-daisyui-hono\.yarn\unplugged\esbuild-npm-0.24.0-1252872327\node_modules\esbuild\lib\main.js:1476:15)
    at C:\template-vike-solid-daisyui-hono\.yarn\unplugged\esbuild-npm-0.24.0-1252872327\node_modules\esbuild\lib\main.js:945:25
    at runOnEndCallbacks (C:\template-vike-solid-daisyui-hono\.yarn\unplugged\esbuild-npm-0.24.0-1252872327\node_modules\esbuild\lib\main.js:1316:45)
    at buildResponseToResult (C:\template-vike-solid-daisyui-hono\.yarn\unplugged\esbuild-npm-0.24.0-1252872327\node_modules\esbuild\lib\main.js:943:7)
    at C:\template-vike-solid-daisyui-hono\.yarn\unplugged\esbuild-npm-0.24.0-1252872327\node_modules\esbuild\lib\main.js:970:16
    at responseCallbacks.<computed> (C:\template-vike-solid-daisyui-hono\.yarn\unplugged\esbuild-npm-0.24.0-1252872327\node_modules\esbuild\lib\main.js:622:9)    
    at handleIncomingPacket (C:\template-vike-solid-daisyui-hono\.yarn\unplugged\esbuild-npm-0.24.0-1252872327\node_modules\esbuild\lib\main.js:677:12)
    at Socket.readFromStdout (C:\template-vike-solid-daisyui-hono\.yarn\unplugged\esbuild-npm-0.24.0-1252872327\node_modules\esbuild\lib\main.js:600:7)
    at Socket.emit (node:events:519:28)
    at addChunk (node:internal/streams/readable:561:12)
    at readableAddChunkPushByteMode (node:internal/streams/readable:512:3)
    at Socket.Readable.push (node:internal/streams/readable:392:5)
    at Pipe.onStreamRead (node:internal/stream_base_commons:189:23) {
  errors: [Getter/Setter],
  warnings: [Getter/Setter],
  code: 'PLUGIN_ERROR',
  plugin: 'vike-node:standalone',
  hook: 'closeBundle'
}
```

**Comment by brillout** [comment]
@rtritto Up for digging into this and creating a PR?

**Comment by brillout** [comment]
Nice digging :dizzy: I think checker whether `path.isAbolsute()` does the trick?

**Comment by rtritto** [comment]
Maybe (depends on the use).
What do think about `process.cwd()` to use as base directory?

**Comment by brillout** [comment]
I ain't sure what you mean. How about you open a PR implementing your proposal and we discuss it over there? Looking forward to it!

**Comment by rtritto** [comment]
I created (draft) a _vite-plugin_ for bundle, maybe can help for this issue: https://github.com/vite-pwa/vite-plugin-pwa/discussions/808

**Comment by rtritto** [comment]
Note: in the `vite-plugin` for bundle, I used `process.cwd()` to get root path

**Comment by rtritto** [comment]
Fixed with:

```diff
export default {
  build: {
    target: 'esnext',
-    outDir: path.resolve(import.meta.dirname, 'dist')
+    outDir: '../dist'
  }
}
```

**Comment by brillout** [comment]
@rtritto We should either support this (I think we can), or at lease improve the error message (i.e. add an `assertUsage()`).

**Comment by magne4000** [comment]
> Nice digging 💫 I think checking whether `path.isAbolsute()` does the trick?

I created a fix in that way.

**Comment by rtritto** [comment]
Still not fixed with [v0.2.10](https://github.com/vikejs/vike-node/tree/v0.2.10)

```diff
// vite.config.ts

export default {
  build: {
    target: 'esnext',
+    outDir: path.resolve(import.meta.dirname, 'dist')
+    OR
+    outDir: path.resolve(import.meta.dirname, 'dist').replaceAll('\\', '/')  // POSIX
  }
}
```
